homeassistant/components/mqtt/discovery.py

```diff
@@ -244,15 +245,15 @@
                     hass, component, platform, payload, hass_config)
                 return
 
-            config_entries_key = '{}.{}'.format(component, platform)
+            config_entries_key = '{}.{}'.format(component, 'mqtt')
```

Comment:
Revert this to `platform`?

Reply:
No, can't be reverted due to the limitations in the config entry system as you pointed out.